File: src/types/chapter.ts
// ---- Chapter ----
export interface Chapter {
  id: string;
  volume_id: string;
  project_id: string;
  title: string;
  order_index: number;
  status: "draft" | "writing" | "final";
  word_count: number;
  summary: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateChapter {
  volume_id: string;
  project_id: string;
  title: string;
}

export interface UpdateChapterMeta {
  title?: string | null;
  status?: "draft" | "writing" | "final" | null;
  summary?: string | null;
}

// ---- Chapter Content ----
export interface ChapterContent {
  chapter_id: string;
  content_json: string;
  content_text: string;
  updated_at: string;
}

File: src-tauri/src/models/chapter.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Chapter {
    pub id: String,
    pub volume_id: String,
    pub project_id: String,
    pub title: String,
    pub order_index: i32,
    pub status: String,
    pub word_count: i64,
    pub summary: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateChapter {
    pub volume_id: String,
    pub project_id: String,
    pub title: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateChapterMeta {
    pub title: Option<String>,
    pub status: Option<String>,
    pub summary: Option<String>,
}
